osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -307,6 +307,17 @@ private ArchiveReader getReaderFrom(string path)
         /// <returns>The imported beatmap, or an existing instance if it is already present.</returns>
         private BeatmapSetInfo importToStorage(ArchiveReader reader)
         {
+            string mapName;
+            // let's make sure there are actually .osu files to import.
+            try
+            {
+                mapName = reader.Filenames.First(f => f.EndsWith(".osu"));
+            }
+            catch
+            {
+                throw new InvalidOperationException("No beatmap files found in the map folder.");

```

Comment:
I found only one line  that call this function https://github.com/ppy/osu/blob/d72a479d9dffa57575ada2b944e04568cc63b3c8/osu.Game/Beatmaps/BeatmapManager.cs#L151